plugins/grid-bookmark/src/GridBookmarkWidget/components/DownloadBookmarks.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react'
+import { observer } from 'mobx-react'
+import { saveAs } from 'file-saver'
+
+import {
+  IconButton,
+  Button,
+  Dialog,
+  DialogTitle,
+  Select,
+  MenuItem,
+  makeStyles,
+} from '@material-ui/core'
+import CloseIcon from '@material-ui/icons/Close'
+import GetAppIcon from '@material-ui/icons/GetApp'
+
+import { GridBookmarkModel } from '../model'
+import { LabeledRegion } from '../types'
+
+function downloadBookmarkFile(
+  bookmarkedRegions: LabeledRegion[],
+  fileFormat: string,
+) {
+  const fileHeader =
+    fileFormat === 'TSV'
+      ? 'chrom\tstart\tend\tlabel\tassembly_name\tcoord_range\n'
+      : ''
+
+  const fileContents = bookmarkedRegions
+    .map(b => {
+      const { label } = b
+      const labelVal = label === '' ? 'NA' : label
+      const locString = `${b.refName}:${b.start}..${b.end}`
+      const bedVal = `${b.refName}\t${b.start}\t${b.end}\t${labelVal}`
+
+      if (fileFormat === 'BED') {
+        return `${bedVal}\n`
+      } else {
+        return `${bedVal}\t${b.assemblyName}\t${locString}\n`
+      }
+    })
+    .reduce((a, b) => a + b, fileHeader)
```

Comment:
could probably use join('') instead of reduce

Reply:
and if interested, use join('\n') and not add a newline inside the map loop

Reply:
I use reduce here because I want the first line to be the file header for either BED or TSV

Reply:
I guess that is valid but my brain doesn't really parse this immediately whereas a join would work for me. I'd use fileContents = fileHeader.concat(...).join('\n')